src/main/java/fm/last/moji/impl/FileUploadOutputStream.java

```diff
@@ -90,12 +92,17 @@ public void close() throws IOException {
           delegate.close();
         } finally {
           try {
-            String message = httpConnection.getResponseMessage();
             int code = httpConnection.getResponseCode();
             if (HttpURLConnection.HTTP_OK != code && HttpURLConnection.HTTP_CREATED != code) {
-              throw new IOException(code + " " + message);
-            } else {
-              log.debug("Status: HTTP {} - {}", code, message);
+              InputStream is;
+              if (200 <= httpConnection.getResponseCode() && httpConnection.getResponseCode() <= 299) {
+                is = httpConnection.getInputStream();
+              } else {
+                is = httpConnection.getErrorStream();
+              }
+              String body = IOUtils.toString(is);
```

Comment:
I think let's not log the entire body for now and just keep the information to the message and the "peer". If we really find a case in the future that requires the body we could add that then but let's not do it for now as it's more complicated and a lot more wordy.